tests/foreach/foreach.t

```diff
@@ -0,0 +1,88 @@
+SETUP TEST DIR
+  $ cp -r $TESTDIR/cram_dir/test_dir_1 test_dir_1
+  $ cd test_dir_1
+  $ git init
+  .* (re)
+  $ git remote add origin file://$TESTDIR/cram_dir/test_dir_1
+  $ git add *
+  $ git commit -m "First Commit"
+  .* (re)
+  .* (re)
+  .* (re)
+  $ mkdir src
+  $ cd src
+  $ cp -r $TESTDIR/cram_dir/test_dir_2 test_dir_2
```

Comment:
I'm still not sold on this approach. Here you are essentially re-creating what `quark up` would do in a sane environment, but with a remote conflicting with what is written in `subprojects.quark`, then do a `quark freeze` (that, again, isn't supported if `quark up` hasn't run) and then check the results.

IMO this is not a good way to do this test, as you set up a facade of a project manually instead of letting it do to `quark`; this may work today, but it's not really part of the contract, as `quark` expects to clone its subprojects by itself, not having them set-up externally. The proper way would be more something like:

- prepare your subprojects "remote repos" somewhere, say `$TESTDIR/repos/test_dir1`, `$TESTDIR/repos/test_dir2`, `$TESTDIR/repos/test_dir3` and `$TESTDIR/repos/test_dir4`;
- in all these, fix the `subprojects.quark` so that they point to the correct absolute `file:///` URLs; a simple `sed` will do;
- now clone the "root" project somewhere else (again, using the `file:///` URL), e.g. under `$TESTDIR/checkout`;
- go in there and do a `quark up`; this should fetch all the subprojects;
- _do not do `quark freeze` yet_! take the opportunity to test that `quark foreach` actually works even with unfrozen repos;
- _now_ you can do `quark freeze` and check that `quark foreach` still works.

Reply:
Changed in https://github.com/comelz/quark/pull/27/commits/90c7866033c2fb62d2ef8d3c0dcde91e2466268c